sungc/deglint.py

```diff
@@ -128,41 +129,141 @@ def check_bandNum_exist(self, bandNums, required_bandNums):
                     )
                 )
 
-    def find_file(self, path, keyword):
+    def find_file(self, keyword):
         """
-        find a file in a directory (path) that
+        find a file in a directory (self.meas_dict)
         has a keyword in its filename
 
         Parameters
         ----------
-        path : str
-            Directory to search
-
         keyword : str
             keyword
 
         Returns
         -------
-        filename : None or str
+        filename : str
             filename
-            if None then file with keyword not found
+
+        Raises
+        ------
+        Exception if filename not found
         """
         filename = None
-        for root, dirs, files in os.walk(path):
-            for f in files:
-                if f.find(keyword) != -1:
-                    filename = pjoin(root, f)
+        for key in self.meas_dict.keys():
```

Comment:
You can loop directly over the keys of a dictionary without having to specify `.keys`.

e.g.

```python
a = {"a": 1, "b": 2, "c": 3}
for key in a:
    print(a)
```